packages/components/forms/src/base-input/BaseInput.styles.ts

```diff
@@ -1,7 +1,26 @@
 import { css } from 'emotion';
 import tokens from '@contentful/f36-tokens';
+import type { CSSObject } from '@emotion/serialize';
 
-const getStyles = ({ as, isDisabled, isInvalid }) => ({
+const sizeToStyles = ({ size }): CSSObject => {
+  switch (size) {
+    case 'small':
+      return {
+        fontSize: tokens.fontSizeM,
+        lineHeight: tokens.lineHeightM,
+        padding: `${tokens.spacing2Xs} ${tokens.spacingXs}`,
+        height: `32px`,
+      };
+    default:
+      return {
+        fontSize: tokens.fontSizeM,
+        lineHeight: tokens.lineHeightM,
+        height: `40px`,
+      };
+  }
+};
+
+const getStyles = ({ as, isDisabled, isInvalid, size }) => ({
```

Comment:
Same suggestion i gave before, since you only have two size variants, the logic in `sizeToStyles` could be moved here and transformed into a if statement

Even if you keep in another function, it's only two cases.... does it need to be a switch if it's only two cases?